hvsock.go

```diff
@@ -279,6 +288,15 @@ func (conn *HvsockConn) CloseWrite() error {
 	return nil
 }
 
+// CloseReadWrite shuts down both the read and write end of the socket, preventing future operations.
+func (conn *HvsockConn) CloseReadWrite() error {
```

Comment:
Is this added because it is being used somewhere in hcsshim? I am mainly asking because the `IsClosed` call will work correctly only when the `HvsockConn` is closed using the `Close` method. But if `CloseReadWrite` is called then `IsClosed` will keep returning false. Unless we have a specific reason to call `CloseReadWrite` instead of `Close` on a socket, let's avoid that.

Reply:
I added it since the MS docs mention it as one of the options when calling shutdown, basically a combo `CloseRead` and `CloseWrite`. But I agree, it is confusing; Ill remove it.